src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -989,6 +1029,48 @@ codet java_bytecode_convert_methodt::convert_instructions(
       statement=std::string(id2string(statement), 0, statement.size()-2);
     }
 
+    // we throw away the first statement in an exception handler
+    // as we don't know if a function call had a normal or exceptional return
+    size_t e;
```

Comment:
Use an iterator instead of the index (and then compare the iterator to end() after that loop)